app/models/api_request.rb

```diff
@@ -0,0 +1,47 @@
+class ApiRequest
+  def query(query)
+    uri = request_uri(query)
+    res = send_api_request(uri)
+    format_response(res)
+  end
+
+  def request_uri(query = nil)
+    URI::HTTPS.build(
+      host: host,
+      path: path,
+      query: query
+    )
+  end
+
+  private
+
+  def format_response(res)
+    {
+      code: res.code,
+      body: JSON.parse(res.body)
+    }.with_indifferent_access
+  end
+
+  # Inheriting class defines
+  def host
+    ""
```

Comment:
[suggestion] should we have this and path throw an error if they're not implemented in the specific classes?

Reply:
I like that idea!  I'll do it for query, too.